ArgumentCountError occurred when calling a mocked function with no arguments

Calling a mocked function using `andImplement` with no arguments raises ArgumentCountError.

staticmock: 4.0.0

```php
public function testFoo(): void
    {
        $mock = \StaticMock::mock(Foo::class)
            ->shouldReceive('foo')
            ->andImplement(
                function () {

                }
            );

        Foo::foo();

        $mock->assert();
    }
```

```console
% ./vendor/bin/phpunit test
PHPUnit 9.5.4 by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)

Time: 00:00.004, Memory: 4.00 MB

There was 1 error:

1) Crhg\StaticMockTest\FooTest::testFoo
ArgumentCountError: Too few arguments to function StaticMock\Recorder\Arguments::record(), 1 passed in /Users/matsui/src/staticmock-test/vendor/tototoshi/staticmock/src/StaticMock/Fake.php on line 57 and exactly 2 expected

/Users/matsui/src/staticmock-test/vendor/tototoshi/staticmock/src/StaticMock/Recorder/Arguments.php:59
/Users/matsui/src/staticmock-test/vendor/tototoshi/staticmock/src/StaticMock/Fake.php:57
/Users/matsui/src/staticmock-test/vendor/tototoshi/staticmock/src/StaticMock/MethodReplacer/MethodInvoker.php:59
/Users/matsui/src/staticmock-test/test/FooTest.php:18

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

--- a/src/StaticMock/Fake.php
+++ b/src/StaticMock/Fake.php
@@ -54,7 +54,7 @@ class Fake {
         $that = $this;
         return function (&...$args) use ($implementation, $that) {
             Counter::getInstance()->increment($that->hash());
-            Arguments::getInstance()->record($that->hash(), ...$args);
+            Arguments::getInstance()->record($that->hash(), $args);
             return $implementation(...$args);
         };
     }

--- a/test/StaticMock/MockTest.php
+++ b/test/StaticMock/MockTest.php
@@ -187,4 +187,12 @@ class MockTest extends \PHPUnit\Framework\TestCase
         $mock->assert();
     }
 
+    public function testAndReturnWithAnonymousFunctionWithNoParam()
+    {
+        $mock = \StaticMock::mock('StaticMock\Car');
+        $mock->shouldReceive('boo')->times(1)->andImplement(function () { return true; });
+        $p = new Person();
+        $p->drive();
+        $mock->assert();
+    }
 }

